bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java

```diff
@@ -310,6 +328,16 @@ public Capabilities getCapabilities(int capabilities) {
         return CAPABILITIES_DATABASE.getOrDefault(capabilities, new Capabilities());
     }
 
+    /**
+     * Return a Capabilities class instance that corresponds to the given 'type' parameter.
+     *
+     * @param type the shade type.
+     * @return corresponding instance of Capabilities class.
+     */
+    public Capabilities getCapabilitiesForType(int type) {
+        return getCapabilities(TYPE_DATABASE.getOrDefault(type, new Type()).getTypeCapabilities());
+    }
+
```

Comment:
Suggestion: Could we encapsulate/hide the "type override" complexity within this class by merging this method with previous method into this signature:
```
public Capabilities getCapabilities(int type, int capabilities)
```

and then simply return capabilities according to type and capabilities? This would move the logic from HDPowerViewShadeHandler into this specialized class.

Reply:
But maybe we can use only the actual data in the property, vs the functional capabilties via that method. Let me look at it again..

Reply:
I made changes. See comment below. So marking this now as resolved.

Reply:
Excellent!